internal/users/locking/testutils.go

```diff
@@ -17,28 +35,163 @@ var overrideLocked atomic.Bool
 func Z_ForTests_OverrideLocking() {
 	testsdetection.MustBeTesting()
 
+	overriddenMu.Lock()
+	defer overriddenMu.Unlock()
+
+	overriddenWriteLockImpl = append(overriddenWriteLockImpl, writeLockImpl)
 	writeLockImpl = func() error {
 		if !overrideLocked.CompareAndSwap(false, true) {
-			return ErrLock
+			return fmt.Errorf("%w: already locked!", ErrLock)
```

Comment:
My IDE complains: "Error string should not be capitalized or end with punctuation mark"

```suggestion
			return fmt.Errorf("%w: already locked", ErrLock)
```